runtime/lua/vim/img/_backend.lua

```diff
@@ -0,0 +1,12 @@
+---@class vim.img.Backend
+---@field render fun(image:vim.img.Image, opts?:vim.img.Backend.RenderOpts)
+
+---@class vim.img.Backend.RenderOpts
+---@field crop? {x:integer, y:integer, width:integer, height:integer} units are pixels
+---@field pos? {row:integer, col:integer} units are cells
+---@field size? {width:integer, height:integer} units are cells
```

Comment:
> RFC: should we use pixels consistently, and maybe provide a pixels_to_cells util function (and cells_to_pixels)?

I saw something like that in `image.nvim`, so it'd be a good idea. I wasn't happy that the units were different between crop and the others. For me personally, I don't know how I'd think of crop in cells, though, as it's in relation to the raw image file whereas `pos` and `size` are in relation to placement on top of neovim.

So two different coordinate spaces; however, happy to switch `crop` to be the same units

Reply:
Maybe we can punt on this for now, because these fields are dict-like. But there is still some ambiguity with the current key names (`crop.width` is pixels, but `size.width` is cells).

Instead, we could future-proof this by requiring an explicit `unit` field (note: below is pseudocode, I didn't check LuaLS syntax):

```
---@class img.Pos
---@field unit:string
---@field x:integer
---@field y:integer

---@class img.Region
---@field pos1: img.Pos
---@field pos2: img.Pos

---@field crop? img.Region
---@field pos?  img.Pos
---@field size? {unit:string, width:integer, height:integer}
```

Note: the above is flirting with https://github.com/neovim/neovim/issues/25509 territory, and I don't want to encumber this PR with that scope. But it'll help to at least reference it and see if it makes sense to stub a similar skeleton here.